akka-testkit/src/test/scala/akka/testkit/AkkaSpec.scala

```diff
@@ -44,17 +44,14 @@ object AkkaSpec {
     ConfigFactory.parseMap(map.asJava)
   }
 
-  def getCallerName(clazz: Class[_]): String = {
-    val s = Thread.currentThread.getStackTrace
+  def getCallerName(): String =
+    Thread.currentThread.getStackTrace
       .map(_.getClassName)
       .drop(1)
       .dropWhile(_.matches("(java.lang.Thread|.*AkkaSpec.*|.*\\.StreamSpec.*|.*MultiNodeSpec.*|.*\\.Abstract.*)"))
-    val reduced = s.lastIndexWhere(_ == clazz.getName) match {
-      case -1 => s
-      case z  => s.drop(z + 1)
-    }
-    reduced.head.replaceFirst(""".*\.""", "").replaceAll("[^a-zA-Z_0-9]", "_")
-  }
+      .head
+      .replaceFirst(""".*\.""", "")
+      .replaceAll("[^a-zA-Z_0-9]", "_")
```

Comment:
So we drop one which is hopefully this method call, then skip `AkkaSpec`, `StreamSpec`, `MultinodeSpec` and any `Abstract` prefix (for abstract subclasses of the former I guess?). and that should bring us to the constructor calling the superconstructor invoking `getCallerName`. 

I guess you verified it gives the expected system names in some tests?

Makes me wonder why we even needed to invoke this before, if we had the `getClass` already.

Reply:
> I guess you verified it gives the expected system names in some tests?

Yes, I did a quick check and for 'typical' tests this gives the expected name

> Makes me wonder why we even needed to invoke this before, if we had the `getClass` already.

Good question, not sure - *seems* to be some old/defunct feature AFAICS

Reply:
Throw a few assertions in some key specs just to fast-fail if it regresses?

Reply:
StackWalker.getCallerClass is the modern API, but maybe not worth it if tests must run on jdk8.

Reply:
We have a slightly different approach in Typed tests, see `TestKitUtils.testNameFromCallStack`. Is that more robust?

I don't think we want to share the same implementation because we should some day remove the dependency between akka-actor-typed-testkit and akka-testkit, but it could be worth comparing.

Reply:
that indeed seems a bit more sophisticated, brought those in line with each other